### ISSUE TITLE
请问下rm 模型训练大概需要什么级别的显卡，需要几张？

### ISSUE THREAD

**Issue by hehebamei** [OpenLLMAI/OpenRLHF#354]
具体实验中，6张4090完全跑不起来啊，用的是微调后的llama3 8B

**Comment by hijkzzz** [comment]
可以开启 adam offload 或者 LoRA

**Comment by hehebamei** [comment]
这些都开启了，请问一下平常再训练rm模型的时候，推荐什么样的硬件呢？我现在6个4090都跑慢了，还得278个小时...

**Comment by hijkzzz** [comment]
一般都是用 A100 以上的GPU 确实 4090 会比较慢，不过 298个小时数据量肯定也不少。
我们的示例脚本 700k 数据量 a100x8 基本1天就可以训练完。
另外开了 LoRA的话一般不需要adam offload

**Comment by hehebamei** [comment]
这对于4090来说还是很有挑战的，我也使用了你们示例的脚本进行训练rm模型。在6×4090中，仍然需要270多个小时...

**Comment by hehebamei** [comment]
set -x 

read -r -d '' training_commands <<EOF
../train_rm.py \
   --save_path ./checkpoint/llama3-8b-rm \
   --save_steps -1 \
   --logging_steps 1 \
   --eval_steps -1 \
   --train_batch_size 126 \
   --micro_train_batch_size 1 \
   --pretrain /openrlhf/OpenLLMAI/Llama-3-8b-sft-mixture \
   --bf16 \
   --max_epochs 1 \
   --max_len 8192 \
   --zero_stage 3 \
   --learning_rate 9e-6 \
   --dataset /openrlhf/OpenLLMAI/preference_dataset_mixture2_and_safe_pku \
   --apply_chat_template \
   --chosen_key chosen \
   --rejected_key rejected \
   --lora_rank 64 \
   --lora_alpha 64 \
   --flash_attn \
   --gradient_checkpointing
EOF
     # --wandb [WANDB_TOKENS] or True (use wandb login command)


if [[ ${1} != "slurm" ]]; then
    deepspeed $training_commands
fi

**Comment by hijkzzz** [comment]
> set -x
> 
> read -r -d '' training_commands <<EOF ../train_rm.py --save_path ./checkpoint/llama3-8b-rm --save_steps -1 --logging_steps 1 --eval_steps -1 --train_batch_size 126 --micro_train_batch_size 1 --pretrain /openrlhf/OpenLLMAI/Llama-3-8b-sft-mixture --bf16 --max_epochs 1 --max_len 8192 --zero_stage 3 --learning_rate 9e-6 --dataset /openrlhf/OpenLLMAI/preference_dataset_mixture2_and_safe_pku --apply_chat_template --chosen_key chosen --rejected_key rejected --lora_rank 64 --lora_alpha 64 --flash_attn --gradient_checkpointing EOF # --wandb [WANDB_TOKENS] or True (use wandb login command)
> 
> if [[ ${1} != "slurm" ]]; then deepspeed $training_commands fi

这个数据集比较大有 550k 然后你设置的长度为 8192 再加上 4090 不支持 NVLINK 所以必然是很慢
只是为了测试的话改成长度 4096 以内好了

**Comment by hehebamei** [comment]
十分感谢，确实24小时就够了

**Comment by 1609685371** [comment]
> 十分感谢，确实24小时就够了

请问你最后的运行参数配置是什么？我这边情况和你一样，唯一不同的是我训的模型是qwen2-7b，仍旧报oom

set -x

read -r -d '' training_commands <<EOF
openrlhf.cli.train_dpo \
   --save_path /home/hech/chen/model/Qwen2-7B-Instruct/dpo_model \
   --save_steps -1 \
   --logging_steps 1 \
   --eval_steps -1 \
   --train_batch_size 126 \
   --micro_train_batch_size 1 \
   --pretrain /home/hech/chen/model/Qwen2-7B-Instruct \
   --bf16 \
   --max_epochs 1 \
   --max_len 2048 \
   --zero_stage 2 \
   --learning_rate 5e-7 \
   --beta 0.1 \
   --dataset /home/hech/chen/dataset/OpenRLHF/preference_dataset_mixture2_and_safe_pku \
   --apply_chat_template \
   --chosen_key chosen \
   --rejected_key rejected \
   --flash_attn \
   --load_checkpoint \
   --gradient_checkpointing \
   --lora_rank 64 \
   --lora_alpha 64
EOF
    # --use_wandb [WANDB_TOKENS] or True (use wandb login command)
    # --ipo [for IPO]
    # --label_smoothing 0.1 [for cDPO]
    # --ref_offload
    # --packing_samples
    # --nll_loss_coef (Regularization with NLL loss)


if [[ ${1} != "slurm" ]]; then
    deepspeed --include localhost:1,2,3,4,6,7 --module $training_commands
fi

**Comment by hehebamei** [comment]
> > 十分感谢，确实24小时就够了
> 
> 请问你最后的运行参数配置是什么？我这边情况和你一样，唯一不同的是我训的模型是qwen2-7b，仍旧报oom
> 
> set -x
> 
> read -r -d '' training_commands <<EOF openrlhf.cli.train_dpo --save_path /home/hech/chen/model/Qwen2-7B-Instruct/dpo_model --save_steps -1 --logging_steps 1 --eval_steps -1 --train_batch_size 126 --micro_train_batch_size 1 --pretrain /home/hech/chen/model/Qwen2-7B-Instruct --bf16 --max_epochs 1 --max_len 2048 --zero_stage 2 --learning_rate 5e-7 --beta 0.1 --dataset /home/hech/chen/dataset/OpenRLHF/preference_dataset_mixture2_and_safe_pku --apply_chat_template --chosen_key chosen --rejected_key rejected --flash_attn --load_checkpoint --gradient_checkpointing --lora_rank 64 --lora_alpha 64 EOF # --use_wandb [WANDB_TOKENS] or True (use wandb login command) # --ipo [for IPO] # --label_smoothing 0.1 [for cDPO] # --ref_offload # --packing_samples # --nll_loss_coef (Regularization with NLL loss)
> 
> if [[ ${1} != "slurm" ]]; then deepspeed --include localhost:1,2,3,4,6,7 --module $training_commands fi

我是改成zero 2 就快多了

**Comment by Alsac** [comment]
2*A100 能做到7b模型的RL微调吗
